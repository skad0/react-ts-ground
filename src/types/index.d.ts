export interface IVideo {
    url: string;
}

export interface IApplicationState {
    currentVideo: IVideo|null;
    videoList: IVideo[];
}
