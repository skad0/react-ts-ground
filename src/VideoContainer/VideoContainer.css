.VideoContainer {
    display: flex;
    justify-content: center;
}
