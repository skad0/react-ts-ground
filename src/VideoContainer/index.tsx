import * as React from 'react';
import ReactPlayer from 'react-player'

import {IVideo} from '../types';

import './VideoContainer.css';

interface IProps {
    video: IVideo;
    onEnded?: () => void
};

class VideoContainer extends React.Component<IProps> {
    constructor(props: IProps) {
        super(props);
    }

    public render() {
        const {video, onEnded} = this.props;

        return (
            <div className="VideoContainer">
                <ReactPlayer
                    url={video.url}
                    controls={true}
                    playing={true}
                    onEnded={onEnded} />
            </div>
        );
    }
}

export default VideoContainer;
