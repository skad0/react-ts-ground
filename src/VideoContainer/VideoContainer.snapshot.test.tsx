import * as React from 'react';
import * as renderer from 'react-test-renderer';

import VideoContainer from './index';

import {IVideo} from '../types';

const video: IVideo = {
    id: 1,
    url: 'https://www.youtube.com/watch?v=OkwplGEUhkM'
};

describe('VideoContainer', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<VideoContainer video={video} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

