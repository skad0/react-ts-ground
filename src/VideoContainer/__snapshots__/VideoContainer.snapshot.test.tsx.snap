// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoContainer renders correctly 1`] = `
<div
  className="VideoContainer"
>
  <div
    style={
      Object {
        "height": "360px",
        "width": "640px",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;
